packages/ragbits-core/src/ragbits/core/embeddings/base.py

```diff
@@ -17,3 +17,24 @@ async def embed_text(self, data: list[str]) -> list[list[float]]:
         Returns:
             List of embeddings for the given strings.
         """
+
+    def image_support(self) -> bool:  # noqa: PLR6301
+        """
+        Check if the model supports image embeddings.
+
+        Returns:
+            True if the model supports image embeddings, False otherwise.
+        """
+        return False
```

Comment:
Maybe we can make it dynamic to reduce boilerplate code. I'd check here if `embed_image` is implemented or not and we can set `image_support` as `@property` here.

Reply:
The logic here may be a lot more complicated. Specifically in many implementations whether the embedder can support embedding an image will depend on the specific model name and version given. For example my version of `LiteLLMEmbeddings` with image embedding support had this implementation of `image_support`:

```
    def image_support(self) -> bool:
        """
        Check if the model supports image embeddings.

        Returns:
            True if the model supports image embeddings, False otherwise.
        """
        # For now only VertexAI multimodal embeddings are supported for multimodal embeddings
        # We should revisit this once LiteLLM provides a unified API for multimodal embeddings
        return self.model in self.SUPPORTED_MULIMODAL_MODELS
```

(Where `SUPPORTED_MULIMODAL_MODELS` is a static property sourced from LiteLLM's `VertexMultimodalEmbedding().SUPPORTED_MULTIMODAL_EMBEDDING_MODELS`)

If it were this simple - that given class either supports images or not - I wouldn't have `image_support` at all and I wouldn't have added `embed_image` to the base embedding class. Instead, I would define a new abstract interface `EmbeddingsWithImageSupport` (inheriting from the base embeddings class) and added `embed_image` only there. I would then check whether an embedder has image support using `isinstance`. Unfortunately, that's out of the question, because whether a class support image embedding or not may be dynamic and depend on arguments to the constructor.

Reply:
Ok, got it